embedded-graphics/src/style/primitive_style.rs

```diff
@@ -14,7 +14,7 @@ use core::convert::TryFrom;
 /// [`PrimitiveStyleBuilder`]: ../style/struct.PrimitiveStyleBuilder.html
 /// [`non_exhaustive`]: https://blog.rust-lang.org/2019/12/19/Rust-1.40.0.html#[non_exhaustive]-structs,-enums,-and-variants
 /// [`primitive_style!`]: ../macro.primitive_style.html
-#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
```

Comment:
Replacing the manual impl with a derive leads to some odd results. I had to pepper some extra `+ Default` trait bounds on `C` and it got messy fast so I gave up. I've swapped the `new()` and `default()` method bodies as I think that makes a bit more sense - `PrimitiveStyle::new()` now calls `PrimitiveStyle::default()` instead of the other way round. Not a huge change either way tbh.

Reply:
Interesting, I thought deriving would work because `Default` is implemented for every `Option<T>` and not only `Option<T>`s where `T: Default`.
I only suggested this as a minor cleanup and I'm OK with leaving the code the way it is now.